src/IB/FEMechanicsExplicitIntegrator.cpp

```diff
@@ -0,0 +1,727 @@
+    batch_vec_ghost_update(d_X_vecs->get(data_time_str), INSERT_VALUES, SCATTER_FORWARD);
+    d_F_vecs->zero("RHS Vector");
+    d_F_vecs->zero("tmp");
+    for (unsigned part = 0; part < d_meshes.size(); ++part)
+    {
+        if (d_static_pressure_part[part])
+        {
+            computeStaticPressure(
+                d_P_vecs->get(data_time_str, part), d_X_vecs->get(data_time_str, part), data_time, part);
+        }
+        assembleInteriorForceDensityRHS(d_F_vecs->get("RHS Vector", part),
+                                        d_X_vecs->get(data_time_str, part),
+                                        d_P_vecs ? &d_P_vecs->get(data_time_str, part) : nullptr,
+                                        data_time,
+                                        part);
+    }
+    batch_vec_ghost_update(d_F_vecs->get("RHS Vector"), ADD_VALUES, SCATTER_REVERSE);
+    for (unsigned part = 0; part < d_meshes.size(); ++part)
+    {
+        d_fe_projectors[part]->computeL2Projection(d_F_vecs->get("solution", part),
+                                                   d_F_vecs->get("RHS Vector", part),
+                                                   FORCE_SYSTEM_NAME,
+                                                   d_use_consistent_mass_matrix,
+                                                   /*close_U*/ false,
+                                                   /*close_F*/ false);
+    }
+    d_F_vecs->copy("solution", { data_time_str });
+}
+
+void
+FEMechanicsExplicitIntegrator::putToDatabase(Pointer<Database> db)
+{
+    FEMechanicsBase::putToDatabase(db);
+    db->putInteger("EXPLICIT_FE_MECHANICS_INTEGRATOR_VERSION", EXPLICIT_FE_MECHANICS_INTEGRATOR_VERSION);
+}
+
+/////////////////////////////// PROTECTED ////////////////////////////////////
+
+void
+FEMechanicsExplicitIntegrator::doInitializeFEEquationSystems()
+{
+    FEMechanicsBase::doInitializeFEEquationSystems();
+    d_system_dof_map_cache.resize(d_meshes.size());
+    const bool from_restart = RestartManager::getManager()->isFromRestart();
+    for (unsigned int part = 0; part < d_meshes.size(); ++part)
+    {
+        EquationSystems& equation_systems = *d_equation_systems[part];
+        IBTK::setup_system_vectors(&equation_systems,
+                                   { COORDS_SYSTEM_NAME, VELOCITY_SYSTEM_NAME },
+                                   { "current", "half", "new" },
+                                   from_restart);
+        IBTK::setup_system_vectors(
+            &equation_systems, { FORCE_SYSTEM_NAME }, { "current", "half", "new", "tmp", "RHS Vector" }, from_restart);
+    }
+}
+
+void
+FEMechanicsExplicitIntegrator::doInitializeFESystemVectors()
+{
+    FEMechanicsBase::doInitializeFESystemVectors();
+    std::vector<EquationSystems*> equation_systems;
+    for (const auto& es : d_equation_systems) equation_systems.push_back(es.get());
+    d_X_vecs.reset(new LibMeshSystemVectors(equation_systems, COORDS_SYSTEM_NAME));
```

Comment:
@drwells: I cannot think of a simple way to avoid having `FEMechanicsExplicitIntegrator` and `IBFEMethod` both implement this, with the only difference being `LibMeshSystemVectors` vs. `LibMeshIBSystemVectors`.

Reply:
(I _can_ think of complicated ways to do this. 😀 )

Reply:
(For instance using lambda expressions.)

Reply:
I don't really see a problem with this approach - those two classes don't know anything about each-other and this class doesn't need any IB data so this looks totally fine to me.